Optimum Propeller Design

francescogervasio opened the pull request:
**GENERAL DESCRIPTION**
Two different versions of the opti_prop function have been released:
Each function provides the a and a' induction vs chi and the dCt and dCp vs non dimensional radius. 
For the Opti_prop_(P/T) functions, the axial induction value w_0 is determined by using the false position method until the difference of the project ct(cp) and the calculated ct(cp) reaches a small percentage value. 
The functions provide plot charts as well as a .txt  output file. A test case function has also been provided in the related folder.

**INPUT**
  N_blade = Number of blades           [ ] 
  R_hub   = Hub radius percentage    [%]                         
  n_rpm   = Revolutions per minutes  [rpm]                       
  h       = Height                                   [m] 
  V_inf   = Asymptotic speed               [m/s]                   
  Ct       = Thrust coefficients  ( Cp = Power coefficients  )               

**OUTPUT**
a and a' induction speed distributions vs chi (omega*r/V_inf)
Cp and Ct distribution vs di adimensional radius

Commit:
Opti_Prop has been created

Commit:
First draft Optimum Propeller Function. 
The function has not been impelemented yet but a first script has been provided.
The input values have been chosen in order to do a preliminary test of the code up to the determination of the pitch distribution.
It's necessary to implement: the iteration cycle, the interface with Xfoil, the user interface.

EttoreSaetta commented on Opti_Prop.m:
```diff
@@ -0,0 +1,113 @@
+clear all; close all; clc;
```
Non dimenticate di inserire l'header ad inizio funzione.

Commit:
new version: the script has been commented and the header has been added as well. The  4th and the 5th steps, according to the bibliography are in progress since the obtained thrust is way below the expected value. The aerodynamic at different section is likely to be obtained with the specific function (xfoil-matlab interface).

Commit:
The first release has been committed. In this version it is possibile to generate a data sheet (with adimensional radius, chord distribution, pitch angle distribution and thrust distribution) for an optimum propeller of given geometry, aerodynamic and thrust. This script generates plots of the main propeller characteristics. User interface and power option has not been impelmented yet. In this release, geometry and aerodynamic characteristics have to be provided by the user; further releases will include links to geometry and aerodynamic generating functions.

Commit:
A Test function has been released as well as 2 optimum propeller functions: Opti_prop_T performs the design of an optimum propeller with fixed T. 
Opti_prop_P performs the design of an optimum propeller with fixed P.  Both functions peovide usefull plots as well as an output file with chord and pith distributions along the non dimensional radius.

Commit:
Opti_prop_T performs the design of an optimum propeller with fixed T. 
Opti_prop_P performs the design of an optimum propeller with fixed P.

Both functions provided useful plots as well as an output .txt file with chord and pitch distributions along the non-dimensional radius.

A test functions has also been provided in order to test the aforementioned functions

Commit:
Opti_prop_T performs the design of an optimum propeller with fixed T. 
Opti_prop_P performs the design of an optimum propeller with fixed P.

Commit:
Opti_prop_T performs the design of an optimum propeller with fixed T. 
Opti_prop_P performs the design of an optimum propeller with fixed P.

Commit:
Merge commit '3986b642ef9d81409edd537e11cbf7153aa790ce'

Commit:
Two different versions of the opti_prop function have been released. Each function provides the a and a' induction vs chi and the dCt and dCp vs non-dimensional radius; each function provides an output .txt file with the aforementioned values as well. 
For the Opti_prop_(P/T)_corr functions, the axial induction value w_0 has been incremented arbitrarily until the difference of the project ct(cp) and the calculated cp reaches a small percentage value.
For the Opti_prop_(P/T)_Newton functions, the axial induction value w_0 is calculated with the Newton-Raphson method. The stop condition is still the one expressed previously over the project ct(cp) and the calculated cp. This second method is still to be perfected.

Commit:
Two different versions of the opti_prop function have been released. Each function provides the a and a' induction vs chi and the dCt and dCp vs non dimensional radius. 
For the Opti_prop_(P/T) functions, the axial induction value w_0 is determined by using the false position method until the difference of the project Ct(Cp) and the calculated ct(cp) reaches a small percentage value. 
The functions provide plot charts as well as a .txt  output file. A test case function has also been provided in the related folder.

francescogervasio committed:
Create README.md

francescogervasio committed:
Update README.md

Commit:
Two different versions of the opti_prop function have been released. Each function provides the a and a' induction vs chi and the dCt and dCp vs non dimensional radius. 
For the Opti_prop_(P/T) functions, the axial induction value w_0 is determined by using the false position method until the difference of the project ct(cp) and the calculated ct(cp) reaches a small percentage value. 
The functions provide plot charts as well as a .txt  output file. A test case function has also been provided in the related folder.

Corrections list:
- a_radim and a_first radim have been deleted
- R_tip has been added as an input value
- chi definition has been modified
- Cp and Ct variables have been modified according to the previous point

Commit:
Two different versions of the opti_prop function have been released. Each function provides the a and a' induction vs chi and the dCt and dCp vs non dimensional radius. 
For the Opti_prop_(P/T) functions, the axial induction value w_0 is determined by using the false position method until the difference of the project ct(cp) and the calculated ct(cp) reaches a small percentage value. 
The functions provide plot charts as well as a .txt  output file. A test case function has also been provided in the related folder.
dCt and dCp functions have been corrected in order to consider the respective distribution values along the non dimensional radius (dCt/dr, dCp/dr). The efficiency calculated with respect to the project advance ratio has been added as an output.

Commit:
Two different versions of the opti_prop function have been released. Each function provides the a and a' induction vs chi and the dCt and dCp vs non dimensional radius. 
For the Opti_prop_(P/T) functions, the axial induction value w_0 is determined by using the false position method until the difference of the project ct(cp) and the calculated ct(cp) reaches a small percentage value. 
The functions provide plot charts as well as a .txt  output file. A test case function has also been provided in the related folder.
dCt and dCp functions have been corrected in order to consider the respective distribution values along the non dimensional radius (dCt/dr, dCp/dr). The efficiency calculated with respect to the project advance ratio has been added as an output.

EttoreSaetta commented on Opti_prop_T.m:
```diff
@@ -0,0 +1,221 @@
+step T or P value that must be confronted with the design T or P value.
+%}
+
+dT_dradim = N_blade*rho*n_rads*R_tip*r_adim.*(1-a_first).*GAMMA;
```
Here you are dividing 2 times for `R_tip` (one for using `r_adim` and one for `dr_adim`).
So, instead of `R_tip` you should use `R_tip^2`.

EttoreSaetta commented on Opti_prop_T.m:
```diff
@@ -0,0 +1,221 @@
+has been provided and plotted.
+%}
+a       = w/V_inf*((chi.^2)./((1+w/V_inf)^2+(chi.^2)));
+a_first = w/V_inf*((1+w/V_inf)./((1+w/V_inf)^2+chi.^2));
```
Also `a` and `a_first` are influenced by `F`, so these two should be multiplied by `F`.

EttoreSaetta commented on Opti_prop_T.m:
```diff
@@ -0,0 +1,221 @@
+w_j = 2*w;
+%}
+lambda = (n_rads*R_tip/V_inf)^-1;
+F      = (2/pi)*acos(exp((N_blade/(2*lambda))*(r_adim-1)));
```
The computation of `F` should be done before `a` and `a_first` because of the previous comment.

EttoreSaetta commented on Opti_prop_T.m:
```diff
@@ -0,0 +1,221 @@
+The non-dimensional aerodynamic optimal load Gamma, already scaled with
+the Prandtl correction function is implemented.
+%} 
+GAMMA = (4*pi.*F.*(r_adim*R_tip).^2.*a_first*n_rads)/N_blade';
```
Once you have multiplied `a_first` for `F`, the `F` is not anymore needed here, because the effect of the Prandtl correction function ha already been considered.

EttoreSaetta reviewed (commented):
Some other corrections.
The same is for the other function.

Commit:
Two different versions of the opti_prop function have been released. Each function provides the a and a' induction vs chi and the dCt and dCp vs non dimensional radius. 
For the Opti_prop_(P/T) functions, the axial induction value w_0 is determined by using the false position method until the difference of the project ct(cp) and the calculated ct(cp) reaches a small percentage value. 
The functions provide plot charts as well as a .txt  output file. A test case function has also been provided in the related folder.
dCt and dCp functions have been corrected in order to consider the respective distribution values along the non dimensional radius (dCt/dr, dCp/dr). The efficiency calculated with respect to the project advance ratio has been added as an output.

Commit:
The documentation has been added in the related folder

EttoreSaetta reviewed (approved):
LGTM
